payjoin-cli/src/app.rs

```diff
@@ -209,14 +209,37 @@ impl App {
             .await??;
 
             println!("Sent fallback transaction");
-            match ctx.process_response(&mut response.into_reader()) {
+            let error = match ctx.process_response(&mut response.into_reader()) {
                 Ok(Some(psbt)) => return Ok(psbt),
-                Ok(None) => std::thread::sleep(std::time::Duration::from_secs(5)),
-                Err(re) => {
-                    println!("{}", re);
-                    log::debug!("{:?}", re);
+                Ok(None) => {
+                    std::thread::sleep(std::time::Duration::from_secs(5));
+                    continue;
                 }
-            }
+                Err(error) => error,
+            };
+            if error.requires_version(&1) {
+                let (req, ctx) = req_ctx.clone().extract_v1().unwrap();
```

Comment:
This clone is needed here because `extract_v1` takes ownership of self and iam not sure its strictly required, could be a future good ticket to investigate